packages/gatsby-theme-docs/src/components/child-sections-nav.js

```diff
@@ -71,27 +152,61 @@ SectionNavigation.propTypes = {
 };
 
 const ChildSectionsNav = (props) => {
+  const [searchValue, setSearchValue] = useState('');
   const pageData = usePageData();
   const sectionToC = skipEmptyLevels(
     findCurrentSection(props.parent, pageData.tableOfContents)
   );
+  const filteredSectionToCList = sectionToC.items.filter(
+    (item) =>
+      !searchValue ||
+      item.title.toLowerCase().includes(searchValue.toLowerCase())
```

Comment:
This line is the actual search behavior.  I would like to consider supporting multi word searches in a minimal way (i.e. not adding a full text library here).  E.g. when typing "cart    update"  matching the "UpdateCart" heading would be my expected behavior.  

`const words = searchValue.toLowerCase().split(/\W+/)`  (split into array by non-word characters)

Then, the lowercase heading title would have to match *all* of the words in any order (i.e. individual checks, many ways to code that available, e.g. a reduce function on the words array). 

An advanced feature would be to allow typos via allowing a certain levenshtein distance, but this would have to be an alternative match to the substring search we have here.  I would rather wait for feedback before going into actual NLP.

Reply:
In the MC we use a library called `fuse.js` for advanced search capabilities.

https://fusejs.io

I would suggest to focus on this as a follow up, which it also helps with "highlighting" the matching search text.

Reply:
I don't think a "smart" NLP lib like fuse or the many others out there is the right behavior here since the primary use case are data type names which an NLP algorithm fails on anyways because ManyCamelCasedWords are just one word for them anyways unless we start throwing an additional custom decompounder at it that we fill with a custom dictionary of our terms (there are no good english decompounders because english does not use compound words).  

A rabbithole I would not want to couple the suggested simple but effective improvement to.

Reply:
This is now the current search behavior...let me know what you think of it and if we covered every use case with it
694dcc1